### PR TITLE
[decomp] Add decomposition for torch.renorm

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #103780
* __->__ #103858
* #103755
* #103765

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @ngimel @yf225 @chenyang78

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -3135,6 +3139,46 @@ def permute(a: TensorLikeType, *dims) -> TensorLikeType:
     return prims.transpose(a, _permutation)
 
 
+@register_decomposition(aten.renorm)
+@out_wrapper()
+def renorm(
+    input: TensorLikeType, p: Union[bool, int, float], dim: int, maxnorm: float
```
Comment (nkaretnikov) [inline torch/_refs/__init__.py:3145]: Nit: `p` and `maxnorm` are both `Scalar`s in the yaml file, but here they have different types.

```
- func: renorm(Tensor self, Scalar p, int dim, Scalar maxnorm) -> Tensor
```

I think `NumberType` from `torch/_prims_common/__init__.py` corresponds to `Scalar`.

File: torch/_refs/__init__.py [hunk 2]
```diff
@@ -3135,6 +3139,46 @@ def permute(a: TensorLikeType, *dims) -> TensorLikeType:
...
+) -> TensorLikeType:
+    utils.check(not isinstance(p, complex), lambda: "renorm: p must be real-valued")
+    utils.check(p > 0, lambda: "renorm: non-positive norm not supported")
+    utils.check(
+        not isinstance(maxnorm, complex), lambda: "renorm: maxnorm must be real-valued"
+    )
+    utils.check(
+        maxnorm >= 0, lambda: f"renorm: expected maxnorm to be >= 0 but got {maxnorm}"
+    )
+    ndim = input.dim()
```
Comment (nkaretnikov) [inline torch/_refs/__init__.py:3155]: Nit: I think Mike told me at some point that`ndim` is preferred over `dim`.

**Review by nkaretnikov — APPROVED** [review]

**Pull request closed without merging**